pandas/io/gbq.py

```diff
@@ -385,16 +385,25 @@ def run_query(self, query):
         _check_google_client_version()
 
         job_collection = self.service.jobs()
-        job_data = {
-            'configuration': {
-                'query': {
-                    'query': query,
-                    'useLegacySql': self.dialect == 'legacy'
-                    # 'allowLargeResults', 'createDisposition',
-                    # 'preserveNulls', destinationTable, useQueryCache
-                }
+
+        job_config = {
+            'query': {
+                'query': query,
+                'useLegacySql': self.dialect == 'legacy'
+                # 'allowLargeResults', 'createDisposition',
+                # 'preserveNulls', destinationTable, useQueryCache
             }
         }
+        configuration = kwargs.get('configuration')
+        if configuration is not None:
+            if 'query' in configuration:
+                job_config['query'].update(configuration['query'])
+            else:
+                job_config = configuration
```

Comment:
@parthea I think there is one problem about throwing exception with ```useLegacySql```. Current logic force ```useLegacySql``` to be specified as ```'legacy'``` or ```'standart'```. 
So do I need throw exception always when I specify ```useLegacySql``` in  ```configuration```? 
I think rewriting this parameter from configuration is better solution (current version).